var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -0,0 +1,161 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import shutil
+import copy
+
+from spack import *
+
+
+class Cp2k(Package):
+    """CP2K is a quantum chemistry and solid state physics software package
+    that can perform atomistic simulations of solid state, liquid, molecular,
+    periodic, material, crystal, and biological systems
+    """
+    homepage = 'http://www.example.com'
+    url = 'https://sourceforge.net/projects/cp2k/files/cp2k-3.0.tar.bz2'
+
+    version('3.0', 'c05bc47335f68597a310b1ed75601d35')
+
+    variant('mpi', default=True, description='Enable MPI support')
+
+    depends_on('python')  # Build dependency
+
+    depends_on('lapack')
+    depends_on('blas')
+    depends_on('fftw')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('scalapack', when='+mpi')
+
+    # TODO : add dependency on libint
+    # TODO : add dependency on libsmm, libxsmm
+    # TODO : add dependency on elpa
+    # TODO : add dependency on CUDA
```

Comment:
I know, but I didn't go through the makefile in arch to see how they want things to be specified... I think I'll add the otional dependencies after this one will be merged.